shared/MiscUtils.hpp

```diff
@@ -39,4 +49,31 @@ namespace Sombrero {
         //zero_________ /      \/      \/
         return graph;
     }
+
+    // Credit to sc2ad for making the sqrt and pow constexpr
+    namespace detail {
+        constexpr float sqrt(float x, float curr, float prev) {
+            return curr == prev
+                ? curr
+                : sqrt(x, 0.5 * (curr + x / curr), curr);
+        }
+    }
+
+
+    constexpr float power(float x, int p) 
```

Comment:
According to sc2ad, this doesn't support negative exponents. Not sure if I'll allow a merge with this

Reply:
made it handle negatives